lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -50,24 +53,157 @@
+
+    The svp_table used in this plugin is imported (see top of file). It is a
+    table of saturated vapour pressures calculated for a range of temperatures.
+    The import also brings in attributes that describe the range of
+    temperatures covered by the table and the increments in the table.
+
+    """
+    def __init__(self, precision=0.005):
         """
         Initialise class.
+
+        Args:
+            precision (float):
+                The precision to which the Newton iterator must converge before
+                returning wet bulb temperatures.
         """
+        self.precision = precision
+        self.maximum_iterations = 20
 
-    def __repr__(self):
-        """Represent the configured plugin instance as a string."""
-        result = ('<Utilities>')
-        return result
+    @staticmethod
+    def _slice_inputs(temperature, relative_humidity, pressure):
+        """Create iterable or iterator over cubes on which to calculate
+        wet bulb temperature"""
+        slices = None
+        try:
+            vertical_coords = [cube.coord(axis='z').name() for cube in
+                               [temperature, relative_humidity, pressure]
+                               if cube.coord_dims(cube.coord(axis='z')) != ()]
+            if not vertical_coords:
+                slices = [(temperature, relative_humidity, pressure)]
+        except iris.exceptions.CoordinateNotFoundError:
+            slices = [(temperature, relative_humidity, pressure)]
+
+        if slices is None:
+            if len(set(vertical_coords)) > 1 or len(vertical_coords) != 3:
+                raise ValueError('WetBulbTemperature: Cubes have differing '
+                                 'vertical coordinates.')
+            level_coord, = set(vertical_coords)
+            temperature_over_levels = temperature.slices_over(level_coord)
+            relative_humidity_over_levels = relative_humidity.slices_over(
+                level_coord)
+            pressure_over_levels = pressure.slices_over(level_coord)
+            slices = zip(temperature_over_levels,
+                         relative_humidity_over_levels, pressure_over_levels)
+
+        return slices
 
     @staticmethod
-    def specific_heat_of_moist_air(mixing_ratio):
+    def _calculate_latent_heat(temperature):
+        """
+        Calculate a temperature adjusted latent heat of condensation for water
+        vapour using the relationship employed by the UM.
+
+        Args:
+            temperature (np.ndarray):
+                Array of air temperatures (K).
+        Returns:
+            np.ndarray:
+                Temperature adjusted latent heat of condensation (J kg-1).
+        """
+        temp_celcius = temperature + consts.ABSOLUTE_ZERO
+        latent_heat = (-1. * consts.LATENT_HEAT_T_DEPENDENCE * temp_celcius +
+                       consts.LH_CONDENSATION_WATER)
+        return latent_heat
+
+    def _calculate_mixing_ratio(self, temperature, pressure):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4811271)